base/common/src/main/java/com/netscape/certsrv/cert/CertEnrollmentRequest.java

```diff
@@ -349,18 +359,209 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    public void toDOM(Document document, Element certEnrollmentRequestElement) {
+
+        super.toDOM(document, certEnrollmentRequestElement);
+
+        if (getProfileId() != null && !getProfileId().isEmpty()) {
+            Element profileIdElement = document.createElement("ProfileID");
+            profileIdElement.appendChild(document.createTextNode(getProfileId()));
+            certEnrollmentRequestElement.appendChild(profileIdElement);
+        }
+
+        if (getServerSideKeygenP12Passwd() != null && !getServerSideKeygenP12Passwd().isEmpty()) {
+            Element getServerSideKeygenP12PasswdElement = document.createElement("ServerSideKeygenP12Passwd");
+            getServerSideKeygenP12PasswdElement.appendChild(document.createTextNode(getServerSideKeygenP12Passwd()));
+            certEnrollmentRequestElement.appendChild(getServerSideKeygenP12PasswdElement);
+        }
+
+        Element renewalElement = document.createElement("Renewal");
+        renewalElement.appendChild(document.createTextNode(String.valueOf(isRenewal())));
+        certEnrollmentRequestElement.appendChild(renewalElement);
+
+        if (getSerialNum() != null) {
+            Element serialNumberElement = document.createElement("SerialNumber");
+            serialNumberElement.appendChild(document.createTextNode(getSerialNum().toHexString()));
+            certEnrollmentRequestElement.appendChild(serialNumberElement);
+        }
+
+        if (getRemoteHost() != null) {
+            Element remoteHostElement = document.createElement("RemoteHost");
+            remoteHostElement.appendChild(document.createTextNode(getRemoteHost()));
+            certEnrollmentRequestElement.appendChild(remoteHostElement);
+        }
+
+        if (getRemoteAddr() != null) {
+            Element remoteAddressElement = document.createElement("RemoteAddress");
+            remoteAddressElement.appendChild(document.createTextNode(getRemoteAddr()));
+            certEnrollmentRequestElement.appendChild(remoteAddressElement);
+        }
+
+        for (ProfileInput input: getInputs()) {
+            certEnrollmentRequestElement.appendChild(input.toDOM(document));
+        }
+
+        for (ProfileOutput output: getOutputs()) {
+
+            certEnrollmentRequestElement.appendChild(output.toDOM(document));
+
+            NodeList nodes = certEnrollmentRequestElement.getElementsByTagName("profileOutput");
+            System.out.println(nodes.getLength());
```

Comment:
I think it is good but for these debug statements, which should drop away when rebasing on #3711.